src/transaction/l1_handler.rs

```diff
@@ -46,17 +46,21 @@ impl L1Handler {
         nonce: Felt252,
         chain_id: Felt252,
         paid_fee_on_l1: Option<Felt252>,
+        hash_value: Option<Felt252>,
     ) -> Result<L1Handler, TransactionError> {
```

Comment:
Please use a different constructor for this. The user should clearly see that mistakes could lead to inconsistencies by simply looking up the name.